File: backend/api/routes/solver.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import List
from api.services.solver_service import solve

router = APIRouter(prefix="/api/solver", tags=["solver"])


class SolverRequest(BaseModel):
    iterations: int
    oop_range: List[str]
    ip_range: List[str]
    oop_stack: int
    ip_stack: int
    pot: int
    flop: List[str]
    oop_contribution: int
    ip_contribution: int

class SolverResponse(BaseModel):
    oop_strategy: List[dict]
    ip_strategy: List[dict]
    iterations: int
    message: str
    

@router.post("/solve", response_model=SolverResponse)
async def solve_game(request: SolverRequest):
    # use solver service to solve game 
    try:
        result = solve(
            oop_range=request.oop_range,
            ip_range=request.ip_range,
            oop_stack=request.oop_stack,
            ip_stack=request.ip_stack,
            oop_contribution=request.oop_contribution,
            ip_contribution=request.ip_contribution,
            pot=request.pot,
            iterations=request.iterations,
            flop=request.flop
        )
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Solver error: {str(e)}")

    return SolverResponse(
        oop_strategy=result['oop_strategy'],
        ip_strategy=result['ip_strategy'],
        iterations=result['iterations'],
        message="Game solved successfully",
    )

File: frontend/src/api/solver.ts
import api from './client';

export interface SolveRequest {
    oop_range: string[];
    ip_range: string[];
    oop_stack: number;
    ip_stack: number;
    oop_contribution: number;
    ip_contribution: number;
    pot: number;
    flop: string[];
}

export interface HandStrategy {
    hand: string;
    fold: number;
    call: number;
    raise: number;
    ev: number;
}

export interface SolveResponse {
    message: string;
    oop_strategy: HandStrategy[];
    ip_strategy: HandStrategy[];
}

export const solverApi = {
    solve: async (request: SolveRequest) => {
        const response = await api.post<SolveResponse>('/solver/solve', request);
        return response.data;                                                       
    }
}